### DISCUSSION TITLE
How to check if models are in sync with generated migrations?

### DISCUSSION THREAD

**Discussion by mateusz-glowinski** [miguelgrinberg/Flask-Migrate#491] [Q&A] [answered]
Hi,
is there a way to check that models are in sync with generated migrations?

[alembic-autogen-check](https://github.com/4Catalyzer/alembic-autogen-check) is a nice tool to check that with `alembic`,
but it isn't working with `flask-migrate`. It gives below error:
```
alembic.util.exc.CommandError: No 'script_location' key found in configuration.
```

I saw [this](https://stackoverflow.com/questions/41343316/how-to-check-if-there-are-pending-migrations-when-using-sqlalchemy-alembic) and [this](https://github.com/miguelgrinberg/Flask-Migrate/issues/354) topic, but those are checking database and migrations sync, not models and migrations sync.

Thanks for help in advance.

**Comment by miguelgrinberg** [comment, marked as answer]
The tool that you referenced does not check for sync between models and migrations, it checks for sync between models and your database.

You can do the same with the `flask db migrate` command. If this command returns an error, then your models are identical to your database tables and there is nothing to migrate. If this command generates a migration file, that means that there are some differences and these will be listed in the generated migration script. If you don't care to keep the generate migration script, you can safely delete it.

**Comment by mateusz-glowinski** [comment]
thank you @miguelgrinberg for clear answer :) 

Ticket/discussion can be closed.

**Comment by OrriMandarin** [comment]
I can see that `alembic check` command has been merged in `flask-migrate` last week which checks if the models and migration are synchronized. 
@miguelgrinberg Do you know when this feature will be released ?

**Reply by miguelgrinberg** [reply]
This is now released. 🎉

**Reply by OrriMandarin** [reply]
Nice thank you 🙏
